src/pycatcher/catch.py

```diff
@@ -222,20 +223,47 @@ def detect_outliers(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
     else:
         df_pandas = df
 
-    # Calculate the length of the time period in years
-    length_year: float = len(df_pandas.index) / 365.25
+    # Ensure the first column is in datetime format and set it as index
+    df_pandas.iloc[:, 0] = pd.to_datetime(df_pandas.iloc[:, 0])
+    df_pandas = df_pandas.set_index(df_pandas.columns[0]).dropna()
 
-    logging.info("Time-series data in years: %.2f", length_year)
+    # Find the time frequency (daily, weekly etc.) and length of the index column
+    inferred_frequency = df_pandas.index.inferred_freq
+    logging.info("Time frequency: ", inferred_frequency)
+
+    length_index = len(df_pandas.index)
+    logging.info("Length of time index: %.2f", length_index)
 
     # If the dataset contains at least 2 years of data, use Seasonal Trend Decomposition
-    if length_year >= 2.0:
-        logging.info("Using seasonal trend decomposition for outlier detection.")
-        return decompose_and_detect(df_pandas)
 
-    # If less than 2 years of data, use Inter Quartile Range (IQR) method
-    else:
-        logging.info("Using IQR method for outlier detection.")
-        return detect_outliers_iqr(df_pandas)
+    # Set parameter for Week check
+    regex_week_check = r'[W-Za-z]'
+
+    match inferred_frequency:
+        case 'D' if length_index >= 730:
+            logging.info("Using seasonal trend decomposition for for outlier detection in day level time-series.")
+            df_outliers = decompose_and_detect(df_pandas)
+            return df_outliers
+        case 'B' if length_index >= 520:
+            logging.info("Using seasonal trend decomposition for outlier detection in business day level time-series.")
+            df_outliers = decompose_and_detect(df_pandas)
+            return df_outliers
+        case 'MS' if length_index >= 24:
+            logging.info("Using seasonal trend decomposition for for outlier detection in month level time-series.")
+            df_outliers = decompose_and_detect(df_pandas)
+            return df_outliers
+        case 'Q' if length_index >= 8:
+            logging.info("Using seasonal trend decomposition for outlier detection in quarter level time-series.")
+            df_outliers = decompose_and_detect(df_pandas)
+            return df_outliers
+        case _:
+            if regex.match(regex_week_check, inferred_frequency) and length_index >= 104:
+                df_outliers = decompose_and_detect(df_pandas)
+                return df_outliers
+            else:
+                # If less than 2 years of data, use Inter Quartile Range (IQR) method
+                logging.info("Using IQR method for outlier detection.")
+                return detect_outliers_iqr(df_pandas)
```

Comment:
I have a couple of questions here:
1. Will using the index length give the right frequency?
2. What if user gives the duplicate dates in the dataset? How does it behave? 🤔